### ISSUE TITLE
Convert Switch into Compound Component

### ISSUE THREAD

**Issue by jaclynjessup** [Workday/canvas-kit#1774]
## 🚀 Feature Proposal

Convert Switch into a compound component.

## Motivation

Switch has a few improvements listed in #387 & #716 
Which would be addressed in converting the component into a compound component

**Comment by bstanton678** [comment]
A11y acceptance criteria: 

- Users can identify whether the switch is ON or OFF when viewed through grayscale color. (Failing WCAG v2.1, 1.4.1 "Use of Color" where the blue BG color is the only visual signifier for ON position.)
- Recommending either "ON / OFF" labels, or symbols / icons. 
- Update ARIA role from "checkbox" to "switch"
- Tabindex shouldn't be necessary when the component is using an <input> element.

**Comment by alanbsmith** [comment]
Switch could still use a design update to meet @williamjstanton's A11y criteria. The role has been updated to `switch` though. But it has been refactored to work with `FormField.Input` making it a compound component. I'm going to close this until we have some capacity from our designers to add labels.
